x-pack/plugin/core/src/test/java/org/elasticsearch/xpack/core/DataTiersUsageTransportActionTests.java

```diff
@@ -42,28 +77,688 @@ public void testCalculateMAD() {
+
+        Map<String, String> tiers = DataTiersUsageTransportAction.tierIndices(indices);
+        assertThat(tiers.size(), equalTo(5));
+        assertThat(tiers.get("hot-1"), equalTo(DataTier.DATA_HOT));
+        assertThat(tiers.get("hot-2"), equalTo(DataTier.DATA_HOT));
+        assertThat(tiers.get("warm-1"), equalTo(DataTier.DATA_WARM));
+        assertThat(tiers.get("cold-1"), equalTo(DataTier.DATA_COLD));
+        assertThat(tiers.get("cold-2"), equalTo(DataTier.DATA_COLD));
+        assertThat(tiers.get("non-tier"), nullValue());
+    }
+
+    public void testCalculateStatsNoTiers() {
+        // Nodes: 0 Tiered Nodes
+        DiscoveryNodes.Builder discoBuilder = DiscoveryNodes.builder();
+        DiscoveryNode leader = newNode(0, DiscoveryNodeRole.MASTER_ROLE);
+        discoBuilder.masterNodeId(leader.getId());
+
+        DiscoveryNode dataNode1 = newNode(1, DiscoveryNodeRole.DATA_ROLE);
+        discoBuilder.add(dataNode1);
+
+        discoBuilder.localNodeId(dataNode1.getId());
+
+        // Indices: 1 Regular index
+        Metadata.Builder metadataBuilder = Metadata.builder();
+        RoutingTable.Builder routingTableBuilder = RoutingTable.builder();
+
+        IndexMetadata index1 = indexMetadata("index_1", 3, 1);
+        metadataBuilder.put(index1, false).generateClusterUuidIfNeeded();
+        {
+            IndexRoutingTable.Builder indexRoutingTableBuilder = IndexRoutingTable.builder(index1.getIndex());
+            routeTestShardToNodes(index1, 0, indexRoutingTableBuilder, dataNode1);
+            routeTestShardToNodes(index1, 1, indexRoutingTableBuilder, dataNode1);
+            routeTestShardToNodes(index1, 2, indexRoutingTableBuilder, dataNode1);
+            routingTableBuilder.add(indexRoutingTableBuilder.build());
+        }
+
+        // Cluster State and create stats responses
+        ClusterState clusterState = ClusterState.builder(new ClusterName("test"))
+            .nodes(discoBuilder)
+            .metadata(metadataBuilder)
+            .routingTable(routingTableBuilder.build())
+            .build();
+
+        long byteSize = randomLongBetween(1024L, 1024L * 1024L * 1024L * 30L); // 1 KB to 30 GB
+        long docCount = randomLongBetween(100L, 100000000L); // one hundred to one hundred million
+        List<NodeStats> nodeStatsList = buildNodeStats(clusterState, byteSize, docCount);
+
+        // Calculate usage
+        Map<String, String> indexByTier = DataTiersUsageTransportAction.tierIndices(clusterState.metadata().indices());
+        Map<String, DataTiersFeatureSetUsage.TierSpecificStats> tierSpecificStats =
+            DataTiersUsageTransportAction.calculateStats(nodeStatsList, indexByTier, clusterState.getRoutingNodes());
+
+        // Verify - No results when no tiers present
+        assertThat(tierSpecificStats.size(), is(0));
+    }
+
+    public void testCalculateStatsTieredNodesOnly() {
+        // Nodes: 0 Tiered Nodes
```

Comment:
I think this comment might be misleading since there are tiered nodes in this test? Or maybe this is in reference to the number of indices?

Reply:
Ah, this is a copy pasta fail from a previous test. Good catch